src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size == 0) {
+            this.head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        this.tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index,size + 1);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> node = findValueByIndex(index);
+            Node<T> newNode;
+            if (node == head) {
+                newNode = new Node<>(null, value, node);
+                head = newNode;
+            } else {
+                newNode = new Node<>(node.prev, value, node);
+                node.prev.next = newNode;
+            }
+            node.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, size);
+        return findValueByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, size);
+        if (index == size) {
+            add(value);
+        }
+        checkIndex(index, size);
+        Node<T> node = findValueByIndex(index);
+        Node<T> newNode;
+        if (size - 1 == index) {
+            newNode = new Node<>(node.prev, value, null);
+            node.prev.next = newNode;
+            tail = newNode;
+        } else if (index == 0) {
+            newNode = new Node<>(null, value, node.next);
+            node.next.prev = newNode;
+            head = newNode;
+        } else {
+            newNode = new Node<>(node.prev, value, node.next);
+            node.next.prev = newNode;
+            node.prev.next = newNode;
+        }
+        return node.value;
```

Comment:
```suggestion
        checkIndex(index, size);
        Node<T> node = findValueByIndex(index);
        T previousValue = node.value;
        node.value = value;
        return previousValue;
```
just set another value in node